<?xml version="1.0"?>

<launch>

  <arg name="robot_id" default="4"/>
  <arg name="target_x" default="2.0"/>
  <arg name="target_y" default="0.75"/>
  <arg name="target_ox" default="0.0"/>
  <arg name="target_oy" default="0.0"/>
  <arg name="target_oz" default="-0.75"/>
  <arg name="target_ow" default="0.66"/>

  <!-- derived args -->
  <arg name="prefix" value="robot$(arg robot_id)"/>
  <arg name="goal_pub_topic" value="/$(arg prefix)/goal"/>

  <!-- Send robot to a precise target -->
  <node pkg="rostopic" type="rostopic" name="robot_goal_pub"
    args="pub $(arg goal_pub_topic) geometry_msgs/PoseStamped '{header: { seq: 0,   stamp: {secs: 0, nsecs: 0}, frame_id: 'world'}, pose: {  position: { x: $(arg target_x), y: $(arg target_y), z: 0.0}, orientation:{ x: $(arg target_ox),  y: $(arg target_oy), z: $(arg target_oz), w: $(arg target_ow)}}}' -1 --latch"
    output="screen"/>


</launch>
